### ISSUE TITLE
Unused Credit Accounts should be deletable

### ISSUE THREAD

**Issue by sbts** [ledgersmb/LedgerSMB#3239]
Currently, if a credit account is accidentally created, you can't delete it without resorting to manual manipulation of the DB.

We should enable a delete option that either isn't shown if there has been activity on the account, or checks for activity and warns the user if it can't be deleted.

Currently the only option to limit a credit accounts availability in the UI is to set a prehistoric end date.

**Comment by ehuelsmann** [comment]
You can enter a historic date in the credit account "End Date" field. That should remove the credit account from the selection drop-downs. With that as a work-around, why do we want to make them deletable (which is *much* more work, because we need to identify conditions where it's safe to present the user with a Delete button)?

**Comment by sbts** [comment]
Fair enough if closed accounts are not listed in searches and dropdown boxes.
However. As discussed. This is currently not the case.

Even closed accounts are listed when searching for contacts

**Comment by ehuelsmann** [comment]
I don't think that the fact that closed accounts are currently incorrectly excluded, should warrant the implementation of a workaround which could be a bigger feature than the original problem.
